lucene/CHANGES.txt

```diff
@@ -156,6 +156,9 @@ Improvements
 * LUCENE-9531: Consolidated CharStream and FastCharStream classes: these have been moved
   from each query parser package to org.apache.lucene.queryparser.charstream (Dawid Weiss).
 
+* LUCENE-8982: Make NativeUnixDirectory pure java with FileChannel direct IO flag,
```

Comment:
Yes according to https://bugs.openjdk.java.net/browse/JDK-8189192, `com.sun.nio.file.ExtendedOpenOption.DIRECT` was added since JDK 10, so this is Lucene 9.0 only feature.